src/fileformats/ocd_file_import.cpp

```diff
@@ -1771,6 +1833,188 @@ void OcdFileImport::setupPointSymbolPattern(PointSymbol* symbol, std::size_t dat
+	    {   0, 192, 128 },
+	    {   0, 192, 192 },
+	    {   0, 192, 255 },
+	    {   0, 255,   0 },
+	    {   0, 255,  64 },
+	    {   0, 255, 128 },
+	    {   0, 255, 192 },
+	    {   0, 255, 255 },
+	    {  64,   0,   0 },
+	    {  64,   0,  64 },
+	    {  64,   0, 128 },
+	    {  64,   0, 192 },
+	    {  64,   0, 255 },
+	    {  64,  64,   0 },
+	    {  64,  64,  64 },
+	    {  64,  64, 128 },
+	    {  64,  64, 192 },
+	    {  64,  64, 255 },
+	    {  64, 128,   0 },
+	    {  64, 128,  64 },
+	    {  64, 128, 128 },
+	    {  64, 128, 192 },
+	    {  64, 128, 255 },
+	    {  64, 192,   0 },
+	    {  64, 192,  64 },
+	    {  64, 192, 128 },
+	    {  64, 192, 192 },
+	    {  64, 192, 255 },
+	    {  64, 255,   0 },
+	    {  64, 255,  64 },
+	    {  64, 255, 128 },
+	    {  64, 255, 192 },
+	    {  64, 255, 255 },
+	    { 128,   0,   0 },
+	    { 128,   0,  64 },
+	    { 128,   0, 128 },
+	    { 128,   0, 192 },
+	    { 128,   0, 255 },
+	    { 128,  64,   0 },
+	    { 128,  64,  64 },
+	    { 128,  64, 128 },
+	    { 128,  64, 192 },
+	    { 128,  64, 255 },
+	    { 128, 128,   0 },
+	    { 128, 128,  64 },
+	    { 128, 128, 128 },
+	    { 128, 128, 192 },
+	    { 128, 128, 255 },
+	    { 128, 192,   0 },
+	    { 128, 192,  64 },
+	    { 128, 192, 128 },
+	    { 128, 192, 192 },
+	    { 128, 192, 255 },
+	    { 128, 255,   0 },
+	    { 128, 255,  64 },
+	    { 128, 255, 128 },
+	    { 128, 255, 192 },
+	    { 128, 255, 255 },
+	    { 192,   0,   0 },
+	    { 192,   0,  64 },
+	    { 192,   0, 128 },
+	    { 192,   0, 192 },
+	    { 192,   0, 255 },
+	    { 192,  64,   0 },
+	    { 192,  64,  64 },
+	    { 192,  64, 128 },
+	    { 192,  64, 192 },
+	    { 192,  64, 255 },
+	    { 192, 128,   0 },
+	    { 192, 128,  64 },
+	    { 192, 128, 128 },
+	    { 192, 128, 192 },
+	    { 192, 128, 255 },
+	    { 192, 192,   0 },
+	    { 192, 192,  64 },
+	    { 192, 192, 128 },
+	    { 192, 192, 192 },
+	    { 192, 192, 255 },
+	    { 192, 255,   0 },
+	    { 192, 255,  64 },
+	    { 192, 255, 128 },
+	    { 192, 255, 192 },
+	    { 192, 255, 255 },
+	    { 255,   0,   0 },
+	    { 255,   0,  64 },
+	    { 255,   0, 128 },
+	    { 255,   0, 192 },
+	    { 255,   0, 255 },
+	    { 255,  64,   0 },
+	    { 255,  64,  64 },
+	    { 255,  64, 128 },
+	    { 255,  64, 192 },
+	    { 255,  64, 255 },
+	    { 255, 128,   0 },
+	    { 255, 128,  64 },
+	    { 255, 128, 128 },
+	    { 255, 128, 192 },
+	    { 255, 128, 255 },
+	    { 255, 192,   0 },
+	    { 255, 192,  64 },
+	    { 255, 192, 128 },
+	    { 255, 192, 192 },
+	    { 255, 192, 255 },
+	    { 255, 255,   0 },
+	    { 255, 255,  64 },
+	    { 255, 255, 128 },
+	    { 255, 255, 192 },
+	    { 255, 255, 255 },
+	};
+	
+	auto icon_bits = icon.bits;
+	constexpr int icon_size = 22;
+	QImage image(icon_size, icon_size, QImage::Format_ARGB32_Premultiplied);
+	for (int y = icon_size - 1; y >= 0; --y)
+	{
+		for (int x = 0; x < icon_size; ++x)
+		{
+			image.setPixelColor(x, y, palette[*(icon_bits++)]);
```

Comment:
`icon_bits` is `quint8` and palette size is 125. Suggest range checking/clipping here.

Reply:
Well spotted.